cmd/terramate/cli/script_run.go

```diff
@@ -88,63 +88,95 @@ func (c *cli) runScript() {
 				logger.Fatal().Err(err).Msg("failed to eval script")
 			}
 
-			for jobNum, j := range evalScript.Jobs {
-				c.output.MsgStdOut("")
-				for cmdNum, cmd := range j.Commands() {
-					printCommand(c.stderr, cmd.Args, st.Dir().String(), jobNum, cmdNum)
-
-					env, err := run.LoadEnv(c.cfg(), st.Stack)
-					if err != nil {
-						logger.Fatal().Err(err).Msg("failed to load env")
+			for jobIdx, job := range evalScript.Jobs {
+				for cmdIdx, cmd := range job.Commands() {
+					exc := RunContext{
+						Stack:        st.Stack,
+						Cmd:          cmd.Args,
+						ScriptIdx:    scriptIdx,
+						ScriptJobIdx: jobIdx,
+						ScriptCmdIdx: cmdIdx,
 					}
 
-					if err := c.executeCommand(cmd.Args, st.Dir().HostPath(c.rootdir()), newEnvironFrom(env)); err != nil {
-						logger.Fatal().Err(err).Msg("unable to execute command")
+					if cmd.Options != nil {
+						exc.CloudSyncDeployment = cmd.Options.CloudSyncDeployment
+						exc.CloudSyncTerraformPlanFile = cmd.Options.CloudSyncTerraformPlan
 					}
+
+					runs = append(runs, exc)
 				}
 			}
 		}
 	}
+
+	c.prepareScriptCloudDeploymentSync(runs)
+
+	isSuccessExit := func(exitCode int) bool {
+		return exitCode == 0
+	}
+
+	err := c.RunAll(runs, isSuccessExit, RunAllOptions{
+		Quiet:           c.parsedArgs.Quiet,
+		DryRun:          c.parsedArgs.Script.Run.DryRun,
+		ScriptRun:       true,
+		ContinueOnError: false,
+	})
+	if err != nil {
+		c.fatal("one or more commands failed", err)
+	}
 }
 
-func (c *cli) executeCommand(cmd []string, wd string, env []string) error {
+func (c *cli) prepareScriptCloudDeploymentSync(runStacks []RunContext) {
 	if c.parsedArgs.Script.Run.DryRun {
-		return nil
+		return
 	}
 
-	newCmd, err := makeCommand(cmd, wd, env, c.stdout, c.stderr)
-	if err != nil {
-		return errors.E(err, "failed to prepare command")
+	var deployRuns []RunContext
+	for _, exc := range runStacks {
+		if exc.CloudSyncDeployment {
+			deployRuns = append(deployRuns, exc)
+		}
 	}
 
-	if err := newCmd.Run(); err != nil {
-		return errors.E(err, "failed to execute command")
+	if len(deployRuns) == 0 {
+		return
 	}
 
-	return nil
-}
+	if !c.prj.isRepo {
+		fatal(errors.E("cloud features require a git repository"))
```

Comment:
Would it make sense to print a warning here instead? e.g.:

```golang
printer.Stderr.Warn("Cloud features require a git repository, will not sync deployment")
```

Reply:
I copied the `run` behaviour here (see `runOnStacks()`), which also fails in this case.

Reply:
Makses sense to keep the behaviour similar to `run`, however I'd suggest we should double check if that is intentional behaviour.

Reply:
We agreed to keep the behaviour like this for now and potentially change it in a later PR.